packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -102,155 +111,144 @@ export const prepareResolverKit = (
-            trace('No pending CCTP transaction found for key:', key);
-            throw Error(
-              `No pending CCTP transaction found matching provided details: ${q(key)}`,
-            );
+        settleTransaction(args: TransactionSettlementOfferArgs) {
+          const { transactionRegistry } = this.state;
+          const { status, txId, rejectionReason } = args;
+          if (!transactionRegistry.has(txId)) {
+            trace('No pending transaction found for key:', txId);
+            throw Error(`No pending transaction found matching: ${q(txId)}`);
           }
-          const registryEntry = cctpTransactionRegistry.get(key);
+
+          const registryEntry = transactionRegistry.get(txId);
+
           switch (status) {
-            case 'success':
+            case TxStatus.SUCCESS:
               trace(
-                'CCTP transaction confirmed - resolving pending operation for key:',
-                key,
+                'Transaction confirmed - resolving pending operation for key:',
+                txId,
               );
               registryEntry.vowKit.resolver.resolve();
-              this.facets.reporter.completePendingTransaction(txId, key);
-              cctpTransactionRegistry.delete(key);
+              this.facets.reporter.completePendingTransaction(
+                txId,
+                TxStatus.SUCCESS,
+              );
+              transactionRegistry.delete(txId);
               return;
-            case 'failed':
+
+            case TxStatus.FAILED:
               trace(
-                'CCTP transaction failed - rejecting pending operation for key:',
-                key,
+                'Transaction failed - rejecting pending operation for key:',
+                txId,
               );
               registryEntry.vowKit.resolver.reject(
-                Error(rejectionReason || 'CCTP transaction failed'),
+                Error(rejectionReason || 'Transaction failed'),
+              );
+              this.facets.reporter.completePendingTransaction(
+                txId,
+                TxStatus.FAILED,
               );
-              this.facets.reporter.completePendingTransaction(txId, key);
-              cctpTransactionRegistry.delete(key);
+              transactionRegistry.delete(txId);
               return;
+
             default:
-              throw Fail`Unexpected status ${q(status)} for CCTP transaction: ${q(key)}`;
+              throw Fail`Unexpected status ${q(status)} for transaction: ${q(txId)}`;
           }
         },
       },
       settlementHandler: {
-        async handle(seat: ZCFSeat, offerArgs: CCTPSettlementOfferArgs) {
-          mustMatch(offerArgs, ResolverOfferArgsShapes.SettleCCTPTransaction);
-          const { txDetails, remoteAxelarChain, txId } = offerArgs;
+        async handle(seat: ZCFSeat, offerArgs: TransactionSettlementOfferArgs) {
+          mustMatch(offerArgs, ResolverOfferArgsShapes.SettleTransaction);
```

Comment:
Now that we're using an exo facet for the handler, we can hoist the argument check up to the interface guard: replace `M.any()` with `ResolverOfferArgsShapes.SettleTransaction` and then skip the `mustMatch` here:

```suggestion
        async handle(seat: ZCFSeat, offerArgs: TransactionSettlementOfferArgs) {
```

I started to suggest:

```suggestion
        async handle(seat: ZCFSeat, offerArgs: CopyRecord) {
          mustMatch(offerArgs, ResolverOfferArgsShapes.SettleTransaction);
```

since the caller, Zoe, only guarantees `CopyRecord`. But we can use an exo interface guard.